Decorator/TextDecorator/EmojiDecorator.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Decorator
+{
+    public class EmojiDecorator : TextFilter
+    {
+        public EmojiDecorator(TextFilter textFilter = null) : base(textFilter) { }
+
+        public override string Filter(string text)
+        {
+            text = base.Filter(text);
+
+            if (text.Contains(":-)"))
+                text = text.Replace(":-)", "U+1F60A");
```

Comment:
Need to replace text to emoji not to its text equivalent :)

Reply:
done